File: cpp-00/ex5/ft_count_harvest_recursive.cpp
#include <iostream>

void ft_count_harvest_recursive() {
    static int day   = 1;
    static int limit = -1;

    if (limit == -1) {
        std::cout << "Days until harvest: ";
        std::cin >> limit;
    }

    if (day > limit) {
        std::cout << "Harvest time!" << std::endl;
        day   = 1;
        limit = -1;
        return;
    }

    std::cout << "Day " << day << std::endl;
    day++;
    ft_count_harvest_recursive();
}

int main(void)
{
    ft_count_harvest_recursive();
    return 0;
}

File: python-00/ex5/ft_count_harvest_recursive.py
#!/usr/bin/env python3


def ft_count_harvest_recursive() -> None:
    """count from 1"""
    days = int(input("Days until harvest: "))

    def count(day, limit):
        if day > limit:
            print("Harvest time!")
            return
        print(f"Day {day}")
        count(day + 1, limit)

    count(1, days)


ft_count_harvest_recursive()
